tests/cpp_unit_tests/test_tap_position_optimizer.cpp

```diff
@@ -2,67 +2,267 @@
 //
 // SPDX-License-Identifier: MPL-2.0
 
+#include <power_grid_model/main_core/input.hpp>
 #include <power_grid_model/optimizer/tap_position_optimizer.hpp>
 
+#include <algorithm>
+#include <ranges>
+
 #include <doctest/doctest.h>
```

Comment:
bad practice to include headers after `STL` because they might overload functionality, resulting in unexpected behaviour.

see also #236 

```suggestion
#include <doctest/doctest.h>

#include <algorithm>
#include <ranges>
```